### ISSUE TITLE
The effect of feature backbone, like Conv-4, Resnet18

### ISSUE THREAD

**Issue by feima0969thu** [wyharveychen/CloserLookFewShot#35]
Thanks for work!

I'm not clear about the effect of feature backbone in the neural network. Can I understand that it extract features and send the features to different models, like matching net and relation net?

And I found that in the relationnet.py, you has defined the RelationModule, it is also some layers that extract features. So, I don't know the relationship between the RelationModule and the feature backbone, like Conv-4, Conv-6, Restnet-18?

Thanks a lot!

**Comment by wyharveychen** [comment]
Hello, the RelationModule are layers after feature backbone layers.
And for the difference of mse loss and softmax loss, from my observation, if your task is to predict 20-way or more rather than 5-way classification, softmax loss would outperform the mse loss in terms of accuracy.

**Comment by feima0969thu** [comment]
OK, I see.

Thanks a lot!

**Comment by feima0969thu** [comment]
Thanks for your reply!

I want to further ask whether the feature backbone, like Conv-4, Conv-6, Restnet-18, is pre-trained on other datasets or trained by yourself?

Thanks a lot!

**Comment by wyharveychen** [comment]
Hello, all are trained by myself from scratch.

**Comment by feima0969thu** [comment]
OK.

Thanks a lot!

**Comment by feima0969thu** [comment]
Hi~

I want to further ask that,

In your paper, you compare different methods on miniImageNet and CUB dataset. 
In the meta-training, you randomly select some samples to form an episode. And there are about 60,000 episodes for 1-shot and 40,000 episodes for 5-shot.

I want ask that is the 60,000-episodes of each method the same? The methods include the matching net, protonet, relationnet and maml.

Or is it a random choice of 60,000 or 40,000 episodes for each method when doing meta-training?

I'm confused about this.

Thanks a lot!

**Comment by wyharveychen** [comment]
Hello, we run 60,000-episodes for 1-shot and 40,000-episodes for 5-shot for each method. However, as we would choose the episode with the best validation accuracy, the reported accuracies may correspond to different episodes. See footnote 2 at the bottom of page 6 in our paper for details.

**Comment by gozetmen** [comment]
> Hello, all are trained by myself from scratch.

Hi @wyharveychen 

If it is possible, can you share your pre-trained models which are trained on Imagenet dataset for the backbones Conv-4 and Conv-6? It helps me a lot since I cannot pre-train the whole dataset from scratch.

Thanks in advance.
